src/components/dialog/dialog.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @function UIDialog
+ * @version 0.0.2
+ * @desc display a Dialog
+ */
+
+export default function UIDialog(userOptions) {
+
+    var DOM,
+        state = {};
+        
+    const defaults = {
+        dialog:         '.js-dialog',
+        openBtn:        '.js-dialog-btn',
+        closeBtn:       '.js-dialog-close-btn',
+        backdrop:       '.js-dialog-backdrop',
+        isModal:        false,
+        showBackdrop:   true
+    };
+
+    // Combine defaults with passed in options
+    const settings = Object.assign(defaults, userOptions);
+    
+    init();
+
+    /**
+    * @function init
+    * @desc Initialises the dialog
+    */ 
+    function init() {
+        
+        // Save all DOM queries for future use
+        DOM = {
+            'page': document.querySelectorAll('body')[0],
+            'dialog': document.querySelectorAll(settings.dialog)[0],
+            'openBtn': document.querySelectorAll(settings.openBtn)[0],
+            'closeBtn': document.querySelectorAll(settings.closeBtn)[0],
+            'backdrop': document.querySelectorAll(settings.backdrop)[0]
+        };
+        
+        // Check if gallery exists, return if not
+        if (DOM.dialog === undefined) { return false; }
```

Comment:
How do we feel about 1 line blocks? Personally I'm not a fan but happy to go with what the majority wants.